scrnaseq.Rmd

```diff
@@ -1378,231 +1435,267 @@ if ((known_markers_n > 0) & any(idx_avgplot)) {
+                                                              xlab="Cluster") 
+  p = patchwork::wrap_plots(p_list, ncol=2) + 
+    patchwork::plot_annotation(title="Violin plot of normalised gene expression data, top 1 marker gene per cluster") +
+    patchwork::plot_layout(guides = "collect") &
+    theme(legend.position="bottom")
+  suppressMessages(p)
 }
-p = patchwork::wrap_plots(p_list, ncol=2) + 
-  patchwork::plot_annotation(title="Ridge plot of raw gene expression counts") + 
-  patchwork::plot_layout(guides = "collect") &
-  theme(legend.position="bottom")
-suppressMessages(p)
-```
-
-### Ridge plots (normalised)
-```{r DEG_plot_ridge_norm, fig.height=fig_height_variable, message=FALSE}
-# Ridge plot of normalised gene expression data
-p_list = Seurat::RidgePlot(sc, features=genes_example, slot="data", combine=FALSE)
-for (i in seq(p_list)) {
-  p_list[[i]] = p_list[[i]] + AddStyle(title=genes_example_titles[i], 
-                                       legend_title="Cell identity", 
-                                       fill=param$col_clusters, 
-                                       ylab="Cluster")
+```
+
+#### Ridge plots (raw)
+```{r degs_plot_ridge_raw, eval=degs_found, fig.height=fig_height_degs, message=FALSE}
+if (degs_found) {
+  # Ridge plot of raw gene expression counts
+  p_list = Seurat::RidgePlot(sc, features=genes_example, assay="RNA", slot="counts", combine=FALSE)
+  for (i in seq(p_list)) {
+    p_list[[i]] = p_list[[i]] + AddStyle(title=genes_example_titles[i], 
+                                         legend_title="Cell identity", 
+                                         fill=param$col_clusters, 
+                                         ylab="Cluster")
+  }
+  p = patchwork::wrap_plots(p_list, ncol=2) + 
+    patchwork::plot_annotation(title="Ridge plot of raw gene expression counts, top 1 marker gene per cluster") + 
+    patchwork::plot_layout(guides = "collect") &
+    theme(legend.position="bottom")
+  suppressMessages(p)
 }
-p = patchwork::wrap_plots(p_list, ncol=2) + 
-  patchwork::plot_annotation(title="Ridge plot of normalised and log-transformed gene expression data") + 
-  patchwork::plot_layout(guides = "collect") &
-  theme(legend.position="bottom")
-suppressMessages(p)
-```
-
-### Dot plot
-```{r DEG_plot_dot}
-# Visualises how feature expression changes across different clusters
-p = Seurat::DotPlot(sc, features=unique(genes_example[length(genes_example):1]), cols=c("lightgrey", param$col)) + 
-  AddStyle(title="Dot plot of normalised gene expression data", 
-           ylab="Cluster") + 
-  theme(axis.text.x = element_text(angle=90, hjust=1, vjust=.5))
-p
 ```
 
-### Heatmap
-```{r DEG_heatmap, fig.height=20}
-# Heatmap of top differentially expressed genes
-top = sc_markers_filt %>% 
-  dplyr::group_by(cluster) %>% 
-  dplyr::top_n(n=10, wt=p_val_adj_score)
-p = Seurat::DoHeatmap(sc, features=top$gene, group.colors=param$col_clusters) + 
```

Comment:
Would it be possible to limit the number of cells per cluster? For a better performance

Reply:
I know this takes a bit, but I think the clients are interested in this very figure, an overview figure of all cells and all degs. So I would like to leave this for now.

Reply:
Added a note that we keep this in mind for later.